dags/openshift_nightlies/scripts/install/rosa.sh

```diff
@@ -137,6 +137,83 @@ _adm_logic_check(){
     echo "Failed to execute oc adm commands after 100 attempts with 5 sec interval" 
 }
 
+_balance_infra(){
+    echo "Initiate migration of prometheus componenets to infra nodepools"
+    oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    oc get sts prometheus-k8s -n openshift-monitoring
+    echo "Restart stateful set pods"
+    oc rollout restart -n openshift-monitoring statefulset/prometheus-k8s 
+    echo "Wait till they are completely restarted"
+    oc rollout status -n openshift-monitoring statefulset/prometheus-k8s
+    echo "Check pods status again and the hosting nodes"
+    oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    for node in $(oc get pods -n openshift-monitoring -o wide | grep -i prometheus-k8s | grep -i running | awk '{print$7}');
+    do
+        if [[ $(oc get nodes | grep infra | awk '{print$1}' | grep $node) != "" ]]; then
+                echo "$node is an infra node"
+        else
+                echo "ERROR: Prometheus-k8s pod on $node is not an infra node"
+                exit 1
+        fi
+    done
+
+    echo "Initiate migration of ingress router-default pods to infra nodepools"
+    echo "Add toleration to use infra nodes"
+    oc patch ingresscontroller -n openshift-ingress-operator default --type merge --patch  '{"spec":{"nodePlacement":{"nodeSelector":{"matchLabels":{"node-role.kubernetes.io/infra":"true"}},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}}}'
```

Comment:
@rsevilla87 I see ingress-perf doing this, should I ignore it here or is ingress-perf going to skip if already applied ?

Reply:
[tested](https://mukrishn-hypershift-airflow.apps.sailplane.perf.lab.eng.rdu2.redhat.com/dags/4.12-rosa-hcp-data-plane/grid?dag_run_id=manual__2023-07-11T12%3A23%3A49.425135%2B00%3A00&task_id=hcp-1-ingress-perf&tab=logs), able to run it. GTM